[WPE] WPE Platform: do not force headless platform to AR24 linear buffer format

carlosgcampos opened the pull request:
#### e4c64c9c92bd4bad8d1dfe9615d6b5dae9de00dc
<pre>
[WPE] WPE Platform: do not force headless platform to AR24 linear buffer format
<a href="https://bugs.webkit.org/show_bug.cgi?id=280263">https://bugs.webkit.org/show_bug.cgi?id=280263</a>

Reviewed by Adrian Perez de Castro.

When used with swrast, shared memory buffers are used with ARGB8888
format, but when using a GPU there&apos;s no reason to force a format. We can
remove the implementation of get_preferred_dma_buf_formats and implement
get_egl_display instead, so that WPEDisplay will build the list of
formats supported by the EGL display.

* Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
(wpeDisplayHeadlessDispose):
(wpeDisplayHeadlessGetEGLDisplay):
(wpe_display_headless_class_init):
(wpeDisplayHeadlessGetPreferredDMABufFormats): Deleted.

Canonical link: <a href="https://commits.webkit.org/284205@main">https://commits.webkit.org/284205@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e31212c3da19307ac77d58b453725986ad46177a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68603 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47995 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21262 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/72672 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19748 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55791 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/19564 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54713 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13135 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43862 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59236 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35175 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40529 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/16659 "Build is in progress. Recent messages:") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18105 "Built successfully") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62481 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; run-api-tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17006 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/74366 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12574 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16258 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62187 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-match-ident.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12614 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59315 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62214 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10156 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3773 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10477 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43796 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/44870 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46064 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44612 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
```diff
@@ -68,16 +91,63 @@ static WPEView* wpeDisplayHeadlessCreateView(WPEDisplay* display)
+        }
+
+        gbm_device_destroy(device);
+        g_set_error(error, WPE_EGL_ERROR, WPE_EGL_ERROR_NOT_AVAILABLE, "Can't get EGL display: failed to create GBM EGL display for %s", filename);
+        return nullptr;
```
Shouldn't this fall-through instead of returning, and try to use the Mesa surfaceless platform as fallback?

carlosgcampos replied:
No, if we have a device we should use GBM, because we can't ensure the device used by the surfaceless platform will be the one we have

aperezdc replied:
Good point!

carlosgcampos committed:
[WPE] WPE Platform: do not force headless platform to AR24 linear buffer format
https://bugs.webkit.org/show_bug.cgi?id=280263

Reviewed by Adrian Perez de Castro.

When used with swrast, shared memory buffers are used with ARGB8888
format, but when using a GPU there's no reason to force a format. We can
remove the implementation of get_preferred_dma_buf_formats and implement
get_egl_display instead, so that WPEDisplay will build the list of
formats supported by the EGL display.

* Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
(wpeDisplayHeadlessDispose):
(wpeDisplayHeadlessGetEGLDisplay):
(wpe_display_headless_class_init):
(wpeDisplayHeadlessGetPreferredDMABufFormats): Deleted.

Canonical link: https://commits.webkit.org/284205@main
